src/si/time.rs

```diff
@@ -44,3 +46,34 @@ quantity! {
         @year: 3.1536_E7; "a", "year", "years";
     }
 }
+
+impl<U, V> From<Time<U, V>> for Duration
+where
+    U: ::si::Units<V> + ?Sized,
+    V: ::num::Num + ::num::AsPrimitive<f64> + ::Conversion<V>,
+{
+    fn from(t: Time<U, V>) -> Duration {
+        let secs = t.value.as_().abs();
```

Comment:
I also tried the obvious thing:

    fn from(t: Time<U, V>) -> Duration {
        let secs = t.get::<second>();
        let nanos = t.get::<nanosecond>();
        Duration::new(secs as u64, nanos as u32)
    }

but I haven't yet found a way to get it to compile.  TBH I'm pretty frustrated with how hard even basic things are.  I try to do a thing, and then I invariably do it wrong because I can't make sense of the current code.  I tried calling from_base, but there were enough unsatisfied trait bound errors that my brain kinda switched off.

It might just be that I'm not good enough to contribute here, because this conversion ought to be the simplest thing in the world to implement.